modules/gis/src/lib/binary-to-geojson.ts

```diff
@@ -139,14 +172,15 @@ function parseFeatureCollection(dataArray): Feature[] {
         // eslint-disable-next-line no-continue
         continue;
       }
-
-      features.push(parseFeature(data, lastIndex, i));
+      const globalFeatureId = data.globalFeatureIds.value[lastIndex];
+      features[globalFeatureId] = parseFeature(data, lastIndex, i);
```

Comment:
Wait... This is not a map - this could index outside of the range. @kylebarron is there a guarantee that `globalFeatureIds` are in [0, features.length -1] and that every feature is covered? Are they just indices into original array? Are all indices guaranteed to be covered?

If so we should create an Array of the appropriate length instead of an empty array on line 159.

@felixpalmer Always risky to sneak in unrelated things in a PR :) One could also have wished for a test case for this.

Reply:
:) I added a test case to `featurecollection.json` ("mixed") which will fail with the old code as the features are returned out of order.

The `globalFeatureIds` are incremented one-by-one as the features are added, so we expect the range to be `[0, features.length -1]`

https://github.com/visgl/loaders.gl/blob/e4c0e56687b9803666ba82914e73375fa424a138/modules/gis/src/lib/geojson-to-binary.ts#L294

Reply:
OK. If you add a code comment clarifying that we depend on that it would help make things clear. 
I am concerned that Array.length may not get set correctly if you are just indexing at random instead of pushing.
Preallocating the array will take care of that and also improve performance.

Reply:
I've added a function to check the range is valid and a couple of tests to verify